### ISSUE TITLE
Correctly terminate mining process

### ISSUE THREAD

**Issue by ManOfFlash** [xmrig/xmrig#308]
I launch xmrig.exe from batch file via task scheduler. When i need full power of my PC i kill a xmrig.exe process by task scheduler. Is correct way of ending process exists?

**Comment by KaiUL** [comment]
Yes there is a proper way to stop it just press the button combination of Ctrl-c in the window it's running in and it will stop!

**Comment by ManOfFlash** [comment]
Problem is that there is no program window when it launched by itself task scheduler (on PC power up event)

**Comment by xmrig** [comment]
Not so good stop process like this, but for CPU miner it's ok, should be no negative effects, except you can fail to allocate huge pages after some time since system start if you use it.
Thank you.

**Comment by ManOfFlash** [comment]
My rig FX-8350 + RX560 working together. 
So i have OpenCL miner launched by this way too. And 1 of 4 times when i kill it with task manager my monitor go black screen (suspend?) and computer totally freezes, power cycle helping (box has no reset button :( ). Yeah, ATI historically make bad VGA drivers but reliable hardware, that not die after 5 years because of humidity inside chip.

**Comment by xmrig** [comment]
Okay, for GPU not property process exit may cause various negative effects for both AMD/NVIDIA cards.
I mark this issue as bug, probably just need add `CTRL_CLOSE_EVENT` handler, need check it.
Thank you.

**Comment by 2010phenix** [comment]
he use over task scheduler...
process is Service us i understand...
need only stop schedule task, no?

**Comment by ManOfFlash** [comment]
my _ByTaskXMRGPU.cmd_:
```
@echo off
set exe=MNR_xmrig-amd.exe
:m1
%~dp0\%exe%
GOTO m1
pause
```
So if exe crashed it being relaunched. (Miner exe can work days or can be found not launched by undefined cause. More investigation needed)
My stop plan
1. Finish task in task manager applet
2. Terminate exe by taskmgr.exe

Yes, task sheduler can be setupped for restarting task, but in this case i must disable task when i need full computer power (or finished task relaunches it self). Accidentally i can forget to enable task after using full PC power. So if computer restarted, disabled miner task not get launched.

**Comment by LTKs1mple** [comment]
You XmRig f*ckers .. your stupid virus infected my goddamn computers... you guys shold get a fucking life if you cant handle this problem... WORTHLESS IDIOTS.

**Comment by mightmay** [comment]
Hi, how can I properly exit XMrig in linux?
Is `Ctrl + C` the proper way of closing XMrig ?
Thanks

**Comment by seantibb** [comment]
I installed xmrig using the powershell script for setting up miners via https://moneroocean.stream/. I needed to use my pc, but xmrig was consuming 96% CPU. I tried killing the xmrig.exe process multiple times in Task Manager, but it just kept respawning. 

As of 13 Nov 2021, the method I found to stop the xmrig.exe process was as follows:

1. Open services.msc
2. Locate the service named "moneroocean_miner"
3. Stop the service

This service is not provided by xmrig.exe, but rather [install path]\nssm.exe. For good measure, I also changed the startup type from Automatic to Manual so I can control this in the future better.

Hope this helps!
